sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpError.cs

```diff
@@ -225,12 +225,15 @@ internal static class AmqpError
                 return new EventHubsException(eventHubsResource, description, EventHubsException.FailureReason.QuotaExceeded, innerException);
             }
 
-            // The link was closed, generally this exception would be thrown for partition specific producers and would be caused by race conditions
-            // between an operation and a request to close a client.
+            // The link was closed; generally this exception would be thrown for partition specific producers and would be caused by race conditions
+            // between an operation and a request to close a client.  It may also occur due to a race condition where the service closes a link or
+            // connection while an operation is being prepared.
+            //
+            // Consider this scenario to be transient so that the operation can be retried if the client itself has not been closed.
 
             if (string.Equals(condition, AmqpErrorCode.IllegalState.Value, StringComparison.InvariantCultureIgnoreCase))
```

Comment:
@JoshLove-msft: This isn't applicable to Service Bus, as it does not explicitly translate this code and, instead, lands in the general communication error bucket which is marked as transient.